libsignal-protocol/src/keys/identity_key_pair.rs

```diff
@@ -30,17 +33,36 @@ impl IdentityKeyPair {
         }
     }
 
-    pub fn public(&self) -> Result<PublicKey, Error> {
```

Comment:
also we don't have to return a `Result` since you are returning `Ok(...)` anyway !

Reply:
Good point! We're already asserting the pointer isn't null, and it *should* be impossible to create an `IdentityKeyPair` which doesn't have a public key so something's gone horribly wrong if that assertion is ever triggered.